orchestrator/careplancontributor/service.go

```diff
@@ -111,57 +111,42 @@ func (s Service) confirm(localFHIR fhirclient.Client, serviceRequestRef string,
 		return nil, fmt.Errorf("failed to create Task: %w", err)
 	}
 
-	// Create channels for polling the task, 1 channel for the result and 1 for an error
-	resultChan := make(chan *fhir.Task)
-	errorChan := make(chan error)
-
 	// Start polling in a new goroutine so that the code continues to the select below
-	go s.pollTaskStatus(*task.Id, resultChan, errorChan)
-
-	// Block until either the task is accepted or an error occurs (error after 10 attempts)
-	select {
-	case result := <-resultChan:
-		log.Info().Msgf("Task/%s accepted: %+v\n", *result.Id, result)
-		enrichedTask, err := s.handleAcceptedTask(task, serviceRequest, &patient)
-		return enrichedTask, err
-	case err := <-errorChan:
-		fmt.Printf("error while polling task %s: %v\n", *task.Id, err)
-		return nil, err
+	err = s.pollTaskStatus(*task.Id)
+
+	if err != nil {
+		return nil, fmt.Errorf("error while polling task %s: %v\n", *task.Id, err)
```

Comment:
Wrap the error with `%w`:
```suggestion
		return nil, fmt.Errorf("error while polling task %s: %w", *task.Id, err)
```